documentation/modules/exploit/linux/local/pihole_remove_commands_lpe.md

```diff
@@ -0,0 +1,288 @@
+## Vulnerable Application
+
+Pi-Hole versions 3.0 - 5.3 allows for command line input to the `removecustomcname` (5.1-5.2.4),
+`removecustomdns` (5.0-5.2.4), and `removestaticdhcp` (3.0-5.2.4) functions without properly validating
+the parameters before passing to `sed`.  When executed as the `www-data` user,
+this alows for a privilege escalation to root since `www-data` is in the
+`sudoers.d/pihole` file with no password.
+
+### Install
+
+More info at [https://github.com/pi-hole/docker-pi-hole/](https://docs.pi-hole.net/)
+
+### Docker v4.3
+
+```
+$ cat ~/docker-compose.yml 
+version: "3"
+
+services:
+  pihole:
+    container_name: pihole
+    image: pihole/pihole:v4.3
+    ports:
+      #- "53:53/tcp"
+      #- "53:53/udp"
+      #- "67:67/udp"
+      - "80:80/tcp"
+    environment:
+      TZ: 'America/Chicago'
+      WEBPASSWORD: ''
+    # Volumes store your data between container upgrades
+    volumes:
+      - './etc-pihole/:/etc/pihole/'
+      - './etc-dnsmasq.d/:/etc/dnsmasq.d/'
+    # Recommended but not required (DHCP needs NET_ADMIN)
+    #   https://github.com/pi-hole/docker-pi-hole#note-on-capabilities
+    cap_add:
+      - NET_ADMIN
+    restart: unless-stopped
+```
+
+Now exploit with `unix/http/pihole_dhcp_mac_exec`
+
+### Docker v5.0
+
+```
+$ cat ~/docker-compose.yml 
+version: "3"
+
+services:
+  pihole:
+    container_name: pihole
+    image: pihole/pihole:v5.0
+    ports:
+      #- "53:53/tcp"
+      #- "53:53/udp"
+      #- "67:67/udp"
+      - "80:80/tcp"
+    environment:
+      TZ: 'America/Chicago'
+      WEBPASSWORD: ''
+    # Volumes store your data between container upgrades
+    volumes:
+      - './etc-pihole/:/etc/pihole/'
+      - './etc-dnsmasq.d/:/etc/dnsmasq.d/'
+    # Recommended but not required (DHCP needs NET_ADMIN)
+    #   https://github.com/pi-hole/docker-pi-hole#note-on-capabilities
+    cap_add:
+      - NET_ADMIN
+    restart: unless-stopped
+```
+
+Get a shell through console:
+
+```
+$ sudo docker-compose exec pihole /bin/bash
+```
+
+Switch to `www-data` user:
+
+```
+su - www-data -s /bin/bash
+```
+
+Now execute a payload to simulate the initial access.
+
+### Docker v5.1
+
+```
+$ cat ~/docker-compose.yml 
+version: "3"
+
+services:
+  pihole:
+    container_name: pihole
+    image: pihole/pihole:v5.1
```

Comment:
Haven't checked; Might be worth checking if this needs the `v` prefix or not too

Reply:
Looks like it's just pihole/pihole:v4.3 that has the issue:
```
docker pull pihole/pihole:v4.3
Error response from daemon: manifest for pihole/pihole:v4.3 not found: manifest unknown: manifest unknown
```

The 5.x versions pull just fine:

```
docker pull pihole/pihole:v5.1
v5.1: Pulling from pihole/pihole
7d2977b12acb: Pull complete 
78cf089748c9: Pull complete 
efe3d34b3c25: Pull complete 
bc429e9f6f2d: Pull complete 
0681d08f984c: Pull complete 
bd2f3a8bdba3: Pull complete 
9da6ac4513e3: Pull complete 
32242121a387: Pull complete 
e5fb100e440d: Pull complete 
b8f4dd2e2aac: Pull complete 
0b422c0e3905: Pull complete 
34204b6c53bf: Pull complete 
63f6f01c8566: Pull complete 
Digest: sha256:bd04f49803b82535bc52afe82b6880cf3c519c23a8f2bf141b6166068b1554f9
Status: Downloaded newer image for pihole/pihole:v5.1
docker.io/pihole/pihole:v5.1
```